src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/DefaultVSCredentialServiceProvider.cs

```diff
@@ -20,41 +22,38 @@ public class DefaultVSCredentialServiceProvider : ICredentialServiceProvider
     {
 
         private readonly Lazy<INuGetUILogger> _outputConsoleLogger;
-        private readonly IServiceProvider _serviceProvider;
+        private readonly IAsyncServiceProvider _asyncServiceProvider;
 
         [ImportingConstructor]
+        internal DefaultVSCredentialServiceProvider(Lazy<INuGetUILogger> outputConsoleLogger)
+            : this(AsyncServiceProvider.GlobalProvider, outputConsoleLogger)
```

Comment:
what is this `AsyncServiceProvider.GlobalProvider` and is it MEF component? how is this being resolved?

Reply:
it's not a MEF component, it's a static property which returns AsyncServiceProvider

Reply:
if it's a static property then why it's part of MEF constructor?

Reply:
it's not in the MEF constructor, it's passed to this() after MEF constructing done.